src/Microsoft.AspNetCore.Mvc.Formatters.Xml/Internal/MvcXmlDataContractSerializerMvcOptionsSetup.cs

```diff
@@ -42,6 +43,16 @@ public void Configure(MvcOptions options)
             options.OutputFormatters.Add(new XmlDataContractSerializerOutputFormatter(_loggerFactory));
             options.InputFormatters.Add(new XmlDataContractSerializerInputFormatter(options));
 
+            // Do not override any user mapping
+            var key = "xml";
+            var mapping = options.FormatterMappings.GetMediaTypeMappingForFormat(key);
```

Comment:
Is the "FormatterMappings" property guaranteed to have value?
If not, please rewrite as:
options.FormatterMappings?.GetMediaTypeMappinForFormat(key)

Reply:
We new up this property when MvcOptions is [new'd up](https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Core/MvcOptions.cs#L38), so unless until a user explicitly sets this to null, it shouldn't be a problem. If a user explicitly sets to null, they are shooting themselves in the foot? (they could clear existing formats if they do not want format mappings)